agent/api/task/task.go

```diff
@@ -709,6 +734,92 @@ func (task *Task) getAllASMSecretRequirements() map[string]apicontainer.Secret {
 	return reqs
 }
 
+// hasLogRouter returns true if the task has a log router container.
+func (task *Task) hasLogRouter() bool {
+	for _, container := range task.Containers {
+		if container.LogRouter != nil { // This is a log router container.
+			return true
+		}
+	}
+	return false
+}
+
+// initializeLogRouterResource initializes the log router task resource and adds it as a dependency of the
+// log router container.
+func (task *Task) initializeLogRouterResource(config *config.Config, resourceFields *taskresource.ResourceFields) error {
+	containerToLogOptions, err := task.collectLogRouterOptions()
+	if err != nil {
+		return errors.Wrap(err, "unable to initialize log router resource")
+	}
+
+	var logRouterResource *logrouter.LogRouterResource
+	for _, container := range task.Containers {
+		logRouter := container.LogRouter
+		if logRouter != nil {
+			var ec2InstanceID string
+			if container.Environment != nil && container.Environment[awsExecutionEnvKey] == ec2ExecutionEnv {
+				ec2InstanceID = resourceFields.EC2InstanceID
+			}
+
+			logRouterResource = logrouter.NewLogRouterResource(config.Cluster, task.Arn, task.Family+":"+task.Version,
+				ec2InstanceID, config.DataDir, logRouter.Type, logRouter.EnableECSLogMetaData, containerToLogOptions)
+			task.AddResource(logrouter.ResourceName, logRouterResource)
+			container.BuildResourceDependency(logRouterResource.GetName(), resourcestatus.ResourceCreated,
+				apicontainerstatus.ContainerCreated)
+			return nil
+		}
+	}
+
+	return errors.New("unable to initialize log router resource because there's no log router container")
+}
+
+// collectLogRouterOptions collects the log options for all the containers that use the log router container
+// as the log driver.
+func (task *Task) collectLogRouterOptions() (map[string]map[string]string, error) {
+	containerToLogOptions := make(map[string]map[string]string)
+
+	for _, container := range task.Containers {
+		if container.DockerConfig.HostConfig == nil {
+			continue
+		}
+
+		hostConfig := &dockercontainer.HostConfig{}
+		err := json.Unmarshal([]byte(*container.DockerConfig.HostConfig), hostConfig)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to decode host config of container %s", container.Name)
+		}
+
+		if hostConfig.LogConfig.Type == logRouterDriverName {
```

Comment:
where is this set?

Reply:
same as above, this comes from task definition

Reply:
customer don't set the hostConfig right? does ACS set this?

Reply:
yes. ACS sets this field to be the value of the logDriver field specified in LogConfiguration object in task definition (https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LogConfiguration.html).